backend/moonstreamapi/providers/moonworm_provider.py

```diff
@@ -0,0 +1,453 @@
+
+        If the given StreamQuery induces filters on this provider, returns those filters. Otherwise, returns
+        None indicating that the StreamQuery does not require any data from this provider.
+        """
+        if query.subscription_types and not any(
+            subtype == self.event_type for subtype in query.subscription_types
+        ):
+            return None
+
+        provider_subscriptions = user_subscriptions.get(self.event_type)
+
+        # If the user has no subscriptions to this event type, we do not have to return any data!
+        if not provider_subscriptions:
+            return None
+        parsed_filters = self.default_filters(provider_subscriptions)
+
+        # from_prefix_length = len("from:")
+        # to_prefix_length = len("to:")
+
+        # subscribed_addresses = {
+        #     subscription.resource_data.get("address")
+        #     for subscription in provider_subscriptions
+        #     if subscription.resource_data.get("address") is not None
+        # }
+
+        # Need check that we can expand logic of parsef filters from query params but it will difficult
+
+        # if query.subscriptions:
+        #     parsed_filters.from_addresses = []
+        #     parsed_filters.to_addresses = []
+        #     for provider_type, raw_filter in query.subscriptions:
+        #         if provider_type != event_type:
+        #             continue
+
+        #         if raw_filter.startswith("from:"):
+        #             address = raw_filter[from_prefix_length:]
+        #             if address in subscribed_addresses:
+        #                 parsed_filters.from_addresses.append(address)
+        #         elif raw_filter.startswith("to:"):
+        #             address = raw_filter[to_prefix_length:]
+        #             if address in subscribed_addresses:
+        #                 parsed_filters.to_addresses.append(address)
+        #         else:
+        #             address = raw_filter
+        #             if address in subscribed_addresses:
+        #                 parsed_filters.from_addresses.append(address)
+        #                 parsed_filters.to_addresses.append(address)
+
+        if not (parsed_filters.addresses):
+            return None
+
+        return parsed_filters
+
+    def stream_boundary_validator(
+        self, stream_boundary: data.StreamBoundary
+    ) -> data.StreamBoundary:
+        """
+        Stream boundary validator for the events provider.
+
+        Checks that stream boundaries do not exceed periods of greater than 24 hours.
+
+        Raises an error for invalid stream boundaries, else returns None.
+        """
+        valid_period_seconds = 24 * 60 * 60
+
+        _, stream_boundary = validate_stream_boundary(
+            stream_boundary, valid_period_seconds, raise_when_invalid=True
+        )
+        return stream_boundary
+
+    def query_events(
```

Comment:
This is a confusing name for a method because `query` is both a noun and a verb. In this order, it is most commonly used as a verb.

When I read `query_events`, I think that you are actually executing a query against the database.

I think a better name for this method would be `generate_events_query`.